### ISSUE TITLE
New Rule: descriptive-link-text

### ISSUE THREAD

**Issue by Grit03** [eslint/markdown#713]
### Rule details

Warns when link text consists entirely of a generic phrase like "click here", "here", "link", or "more" that doesn't describe the link's destination and purpose

### What type of rule is this?

Warns about a potential problem

### Example code

```markdown
<!-- Incorrect Cases -->

[click here](https://eslint.org/docs/latest/)

[here](https://eslint.org/docs/latest/)

[link](https://eslint.org/docs/latest/)

[more](https://eslint.org/docs/latest/)

[Click Here](https://eslint.org/docs/latest/)

[**click here**](https://eslint.org/docs/latest/)

[here][eslint-docs]

[eslint-docs]: https://eslint.org/docs/latest/




<!-- Correct Cases -->

[installation guide](https://example.com/docs)

See the [ESLint documentation](https://eslint.org/docs) for details.

[Click here to read the installation guide](https://example.com/docs)
```

### Prior Art

markdownlint: [MD059 / descriptive-link-text](https://github.com/DavidAnson/markdownlint/blob/main/doc/md059.md)

added in `v0.38.0` and enabled by default. It reports links whose text is one of a configurable list of prohibited phrases (default: `["click here", "here", "link", "more"]`).

### Participation

- [x] I am willing to submit a pull request to implement this rule.

### AI acknowledgment

- [ ] I did not use AI to generate this issue report.
- [x] (If the above is not checked) I have reviewed the AI-generated content before submitting.

### Additional comments

Generic link text is an accessibility problem: screen reader users often navigate a page by pulling up a list of its links out of context, and a list full of "here" and "click here" gives them no way to tell where each link leads. Descriptive link text is required by [WCAG 2.4.4 (Link Purpose)](https://www.w3.org/WAI/WCAG22/Understanding/link-purpose-in-context.html). This would follow the same accessibility track as `require-alt-text` (#365).

Like [MD059](https://github.com/DavidAnson/markdownlint/blob/main/doc/md059.md), the rule would handle non-English phrases by letting users add them to the configurable prohibited list, rather than shipping per-language defaults.

**On naming:** `descriptive-link-text` matches the markdownlint rule name, but `no-generic-link-text` might fit this plugin's naming conventions better. No strong preference either way.

Disclosure: I'm a participant of [open source contribution program OSSCA](https://github.com/eslint-ossca)

**Comment by lumirlumir** [comment]
> Disclosure: I’m a participant in the [Open Source Contribution Program OSSCA](https://github.com/eslint-ossca): confirmed.

Thanks for the issue, but I have two open questions about it:

- Should this rule be considered stylistic? I’m not entirely convinced that it is.
- If this rule isn’t stylistic, I think its name would be better aligned with the `no-restricted-xxx` rules in ESLint core, such as [`no-restricted-imports`](https://eslint.org/docs/latest/rules/no-restricted-imports). For example, we could name it something similar to `no-restricted-link-text`.

If this rule isn’t a stylistic choice, I’m in favor of adding it since it originated in `markdownlint`. I’ll leave this issue open to gather more feedback.
